src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java

```diff
@@ -179,12 +188,34 @@ private void ingestActionsPerUser() {
 
     private void ingestBillPayUsers() {
         if (this.globalProperties.getBoolean(PROPERTY_INGEST_BILLPAY)) {
-            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport()
-                .stream()
-                .map(LegalEntityWithUsers::getUserExternalIds)
-                .flatMap(List::stream)
-                .collect(Collectors.toList())
-                .forEach(this.billpayConfigurator::ingestBillPayUser);
+            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport().forEach((le) -> {
+
+                List<Account> accounts = new ArrayList<>();
+                String externalUserId = le.getUserExternalIds().get(0);
+                loginRestClient.login(externalUserId, externalUserId);
+                String externalLegalEntityId = this.userPresentationRestClient.
+                    retrieveLegalEntityByExternalUserId(externalUserId)
+                    .getExternalId();
+                List<ArrangementItem> arrangementItems = accountsIntegrationRestClient
+                    .getArrangements(externalLegalEntityId);
+
+                arrangementItems.forEach((arrangement) -> {
+                    accounts.add(mapBillPayAccount(arrangement));
+                });
+                le.getUserExternalIds().forEach((user) -> {
+                    billpayConfigurator.ingestBillPayUser(user);
+                    if (this.globalProperties.getBoolean(PROPERTY_INGEST_BILLPAY_ACCOUNTS)) {
+                        billpayConfigurator.ingestBillPayAccounts(user, accounts);
+                    }
+                });
+            });
         }
     }
+
+    private Account mapBillPayAccount(ArrangementItem arrangement) {
```

Comment:
do you wish to map all possible types? Or only `"Current Account"` types?

Reply:
Due to the way billlpay mockbank and the endpoint is, the type of the account doesn't really matter, however to add more coverage i will set the account type as a random account type so it changes

Reply:
you could use the `ArrangementItem` product type to map a sensible type